From cb682f5c51c1924c8279765c04ca101a68416e71 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Wed, 24 Sep 2014 12:41:43 -0400
Subject: [fs] ext4: fix wrong assert in ext4_mb_normalize_request()

Message-id: <1411562503-15088-1-git-send-email-lczerner@redhat.com>
Patchwork-id: 96527
O-Subject: [RHEL 7.1 PATCH] ext4: fix wrong assert in ext4_mb_normalize_request()
Bugzilla: 1146046
RH-Acked-by: Carlos Maiolino <cmaiolino@redhat.com>

From: Maurizio Lombardi <mlombard@redhat.com>

BZ 1146046
https://bugzilla.redhat.com/show_bug.cgi?id=1146046

BREW 800952
http://brewweb.devel.redhat.com/brew/taskinfo?taskID=800952

Upstream commit b5b60778558cafad17bbcbf63e0310bd3c68eb17

 The variable "size" is expressed as number of blocks and not as
 number of clusters, this could trigger a kernel panic when using
 ext4 with the size of a cluster different from the size of a block.

 Cc: stable@vger.kernel.org
 Signed-off-by: Maurizio Lombardi <mlombard@redhat.com>
 Signed-off-by: Theodore Ts'o <tytso@mit.edu>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/mballoc.c b/fs/ext4/mballoc.c
index f2dd470..37931d8 100644
--- a/fs/ext4/mballoc.c
+++ b/fs/ext4/mballoc.c
@@ -3155,7 +3155,7 @@ ext4_mb_normalize_request(struct ext4_allocation_context *ac,
     (unsigned long) ac->ac_o_ex.fe_logical);
   BUG();
  }
- BUG_ON(size <= 0 || size > EXT4_CLUSTERS_PER_GROUP(ac->ac_sb));
+ BUG_ON(size <= 0 || size > EXT4_BLOCKS_PER_GROUP(ac->ac_sb));
 
  /* now prepare goal request */
 
-- 
1.7.1